kernel/src/process_loading.rs

```diff
@@ -442,9 +447,13 @@ pub trait ProcessLoadingAsyncClient {
 /// Various process loaders may exist. This includes a loader from a MCU's
 /// integrated flash, or a loader from an external flash chip.
 pub trait ProcessLoadingAsync<'a> {
-    /// Set the client to receive callbacks about process loading and when
+    /// Set the boot client to receive callbacks about process loading and when
+    /// process loading has finished.
+    fn set_boot_client(&self, client: &'a dyn ProcessLoadingAsyncClient);
+
+    /// Set the runtime client to receive callbacks about process loading and when
     /// process loading has finished.
-    fn set_client(&self, client: &'a dyn ProcessLoadingAsyncClient);
+    fn set_runtime_client(&self, client: &'a dyn ProcessLoadingAsyncClient);
```

Comment:
It just occurred to me we don't actually need to change this trait. It might be confusing that there are two set_client functions, but there is no way via this trait to cause the runtime client to trigger. We should probably just move the set_runtime_client to a pub function and not modify this trait.